jaxrs-testing/src/test/java/com/facebook/airlift/jaxrs/testing/TestThriftBodyGenerator.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2010 Proofpoint, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.airlift.jaxrs.testing;
+
+import com.facebook.airlift.http.client.BodyGenerator;
+import com.facebook.airlift.jaxrs.thrift.TOutputStreamTransport;
+import com.facebook.drift.codec.ThriftCodec;
+import com.facebook.drift.protocol.TFacebookCompactProtocol;
+
+import java.io.OutputStream;
+
+public class TestThriftBodyGenerator<T>
+        implements BodyGenerator
+{
+    private final ThriftCodec<T> thriftCodec;
+    private final T instance;
+
+    public TestThriftBodyGenerator(ThriftCodec<T> thriftCodec, T instance)
+    {
+        this.thriftCodec = thriftCodec;
+        this.instance = instance;
+    }
+
+    @Override
+    public void write(OutputStream out)
+            throws Exception
+    {
+        thriftCodec.write(instance, new TFacebookCompactProtocol(new TOutputStreamTransport(out)));
```

Comment:
Ideally protocol and transport must be specified in the constructor (with some possible defaults).

Reply:
You can use the `com.facebook.drift.transport.netty.codec.Protocol` enum and the `com.facebook.drift.transport.netty.codec.Transport` enum